src/callbacks/get_callbacks.jl

```diff
@@ -175,20 +175,64 @@ function get_diagnostics(parsed_args, atmos_model, Y, p, sim_info, output_dir)
     # Flatten the array of arrays of diagnostics
     diagnostics = vcat(diagnostics_ragged...)
 
+    t_span =
+        dt isa ITime ? ITime(time_to_seconds(parsed_args["t_end"])) - t_start :
+        FT(time_to_seconds(parsed_args["t_end"]) - t_start)
+
     if parsed_args["output_default_diagnostics"]
         diagnostics = [
             CAD.default_diagnostics(
                 atmos_model,
-                sim_info.dt isa ITime ?
-                ITime(time_to_seconds(parsed_args["t_end"])) - t_start :
-                FT(time_to_seconds(parsed_args["t_end"]) - t_start),
+                t_span,
                 start_date;
                 output_writer = netcdf_writer,
             )...,
             diagnostics...,
         ]
     end
-    diagnostics = collect(diagnostics)
+    if parsed_args["output_default_diagnostics"] && (
+        parsed_args["use_exact_jacobian"] ||
+        parsed_args["debug_approximate_jacobian"]
+    )
+        t_span_seconds = dt isa ITime ? seconds(t_span) : t_span
```

Comment:
You can write 
```suggestion
        t_span_seconds = float(t_span)
```
this will work for both Float and ITime